src/Operations/ConvertContoursToMeshes.cc

```diff
@@ -272,20 +272,25 @@ bool ConvertContoursToMeshes(Drover &DICOM_data,
 
             // Identify whether there are adjacent planes within the contour spacing on either side.
             auto l_cp_it = std::prev(m_cp_it);
-            //auto h_cp_it = std::next(m_cp_it);
+            auto h_cp_it = std::next(m_cp_it);
 
             if(l_cp_it == std::cend(ucps)){
                 continue;
             }else if(l_cp_it != std::cend(ucps)){
                 const auto l_cp_dist = std::abs(m_cp_it->Get_Signed_Distance_To_Point(l_cp_it->R_0));
                 if((1.5 * contour_sep) < l_cp_dist) l_cp_it = std::cend(ucps);
             }
-            //if(h_cp_it != std::cend(ucps)){
-            //    const auto h_cp_dist = std::abs(m_cp_it->Get_Signed_Distance_To_Point(h_cp_it->R_0));
-            //    if((1.5 * contour_sep) < h_cp_dist) h_cp_it = std::cend(ucps);
-            //}
+            if(h_cp_it != std::cend(ucps)){
+               const auto h_cp_dist = std::abs(m_cp_it->Get_Signed_Distance_To_Point(h_cp_it->R_0));
+               if((1.5 * contour_sep) < h_cp_dist) h_cp_it = std::cend(ucps);
+            }
 
             auto l_cops = locate_contours_on_plane(*l_cp_it);
+            auto h_cops = locate_contours_on_plane(*h_cp_it);
+
+            decltype(l_cops) m_cops_to_roof_cap;
```

Comment:
m_cops_to_roof_cap = list of reference wrappers to all the contours that need a roof cap (close hole in roof).
amalgamated_contours = list of contours (without reference wrappers) that were created in the for loop.
- need this structure to change the scope of the contours created in the for loop, which then can be properly reference wrappered in m_cops_to_roof_cap

Not sure if there is a better way to do this?

Reply:
ignore, simplified